src/legacy/ui/public/agg_types/controls/top_aggregate.test.tsx

```diff
@@ -0,0 +1,186 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
+import { TopAggregateParamEditor } from './top_aggregate';
+import { OptionedValueProp } from '../param_types/optioned';
+
+describe('TopAggregateParamEditor', () => {
+  let agg: any;
+  let aggParam: any;
+  let defaultProps: any;
+  let options: any;
+
+  beforeEach(() => {
+    options = [
+      {
+        text: 'Min',
+        isCompatible: jest.fn((aggr: any) => aggr.params.field.type === 'number'),
+        disabled: true,
+        value: 'min',
+      },
+      {
+        text: 'Max',
+        isCompatible: jest.fn((aggr: any) => aggr.params.field.type === 'number'),
+        disabled: true,
+        value: 'max',
+      },
+      {
+        text: 'Average',
+        isCompatible: jest.fn((aggr: any) => aggr.params.field.type === 'string'),
+        disabled: true,
+        value: 'average',
+      },
+    ];
+    Object.defineProperty(options, 'byValue', {
+      get: () =>
+        options.reduce((acc: { [key: string]: any }, option: OptionedValueProp) => {
+          acc[option.value] = { ...option };
+
+          return acc;
+        }, {}),
+    });
+    aggParam = {
+      options,
+    };
+    agg = {
+      params: {
+        field: {
+          type: 'number',
+        },
+      },
+      getAggParams: jest.fn(() => [{ name: 'aggregate', options }]),
+    };
+    defaultProps = {
+      agg,
+      aggParam,
+      value: '',
+      showValidation: false,
+      setValue: jest.fn(),
+      setValidity: jest.fn(),
+      setTouched: jest.fn(),
+    };
+  });
+
+  it('should init with the default set of props', () => {
+    const comp = shallowWithIntl(<TopAggregateParamEditor {...defaultProps} />);
+
+    expect(comp).toMatchSnapshot();
+  });
+
+  it('should be disabled if a field type is set but there are no compatible options', () => {
+    options = [];
+    const comp = mountWithIntl(<TopAggregateParamEditor {...defaultProps} showValidation={true} />);
+    const select = comp.find('select');
+
+    expect(defaultProps.setValidity).toHaveBeenCalledWith(true);
+    expect(comp.children().props()).toHaveProperty('isInvalid', false);
+    expect(select.children()).toHaveLength(1);
+    expect(select.props()).toHaveProperty('disabled', true);
+  });
+
+  it('should change its validity due to passed props', () => {
+    const comp = mountWithIntl(
+      <TopAggregateParamEditor {...defaultProps} value={{ value: 'min' }} />
+    );
+
+    expect(defaultProps.setValidity).toHaveBeenCalledWith(true);
+
+    comp.setProps({ showValidation: true, value: undefined });
+
```

Comment:
I'd add a check that the form row is invalid:
```ts
expect(comp.children().props()).toHaveProperty('isInvalid', false);
```

Reply:
done